airflow/dag_processing/processor.py

```diff
@@ -16,610 +16,228 @@
-                self._process.join(timeout=5)
-                if self._process.is_alive():
-                    # Didn't shut down cleanly - kill it
-                    self._kill_process()
-
-        if not self._process.is_alive():
-            self._done = True
-            self.log.debug("Waiting for %s", self._process)
-            self._process.join()
-            self._parent_channel.close()
-            return True
-
-        return False
 
-    @property
-    def result(self) -> tuple[int, int, int] | None:
-        """Result of running ``DagFileProcessor.process_file()``."""
-        if not self.done:
-            raise AirflowException("Tried to get the result before it's done!")
-        return self._result
-
-    @property
-    def start_time(self) -> datetime:
-        """Time when this started to process the file."""
-        if self._start_time is None:
-            raise AirflowException("Tried to get start time before it started!")
-        return self._start_time
-
-    @property
-    def waitable_handle(self):
-        return self._process.sentinel
-
-    def import_modules(self, file_path: str | Iterable[str]):
-        def _import_modules(filepath):
-            for module in iter_airflow_imports(filepath):
-                try:
-                    importlib.import_module(module)
-                except Exception as e:
-                    # only log as warning because an error here is not preventing anything from working, and
-                    # if it's serious, it's going to be surfaced to the user when the dag is actually parsed.
-                    self.log.warning(
-                        "Error when trying to pre-import module '%s' found in %s: %s",
-                        module,
-                        file_path,
-                        e,
-                    )
-
-        if isinstance(file_path, str):
-            _import_modules(file_path)
-        elif isinstance(file_path, Iterable):
-            for path in file_path:
-                _import_modules(path)
-
-
-class DagFileProcessor(LoggingMixin):
+@attrs.define()
+class DagFileProcessorProcess(WatchedSubprocess):
     """
-    Process a Python file containing Airflow DAGs.
-
-    This includes:
-
-    1. Execute the file and look for DAG objects in the namespace.
-    2. Execute any Callbacks if passed to DagFileProcessor.process_file
-    3. Serialize the DAGs and save it to DB (or update existing record in the DB).
-    4. Record any errors importing the file into ORM
+    Parses dags with Task SDK API.
 
-    Returns a tuple of 'number of dags found' and 'the count of import errors'
-
-    :param log: Logger to save the processing process
+    Since DAGs are written with the Task SDK, we need to parse them in a task SDK process such that
+    we can use the Task SDK definitions when serializing. This prevents potential conflicts with classes
+    in core Airflow.
```

Comment:
```suggestion
    Since DAGs are written with the Task SDK, we need to parse them in a task SDK process such that
    we can use the Task SDK definitions when serializing. This class can be thought of as a supervisor that launches a subprocess to parse DAGfiles. This prevents potential conflicts with classes
    in core Airflow.
```